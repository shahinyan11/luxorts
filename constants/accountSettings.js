import ROUTES from './routes';

// eslint-disable-next-line import/prefer-default-export
export const MENU_ITEMS = [
  {
    href: ROUTES.ACCOUNT_SETTINGS.PERSONAL_INFORMATION.PATH,
    icon: 'profile',
    title: { id: 'personalInformation.title' },
    description: { id: 'shared.providePersonalDetailsAndHowWeCanReachYou' },
  },
  {
    href: ROUTES.ACCOUNT_SETTINGS.LOGIN_AND_SECURITY.PATH,
    icon: 'user-shield',
    title: { id: 'shared.loginSecurity' },
    description: { id: 'shared.updateYourPasswordAndSecureYourAccount' },
  },
  {
    href: '#',
    icon: 'bank-card',
    title: { id: 'shared.paymentsPayouts' },
    description: { id: 'shared.reviewPaymentsPayouts' },
  },
  {
    href: ROUTES.ACCOUNT_SETTINGS.INVITE_FRIENDS.PATH,
    icon: 'guests',
    title: { id: 'shared.inviteFriends' },
    description: { id: 'shared.inviteFriendsToJoinLuxortsCommunity' },
  },
];

export const GENDER_TYPE = {
  MALE: 'Male',
  FEMALE: 'Female',
  OTHER: 'Other',
};

export const GENDER_OPTIONS = [
  { value: GENDER_TYPE.MALE, label: { id: 'shared.male' } },
  { value: GENDER_TYPE.FEMALE, label: { id: 'shared.female' } },
  { value: GENDER_TYPE.OTHER, label: { id: 'shared.other' } },
];
