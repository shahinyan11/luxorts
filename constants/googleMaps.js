export const GOOGLE_MAPS_API_KEY = process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY;

export const DEFAULT_LANGUAGE_CODE = 'en';

export const NEW_YORK_COORDS = [-73.97447621784085, 40.764468117826794];

export const GOOGLE_MAPS_ADDRESS_TYPE = {
  STREET_NUMBER: 'street_number',
  ROUTE: 'route',
  LOCALITY: 'locality',
  ADMINISTRATIVE_AREA_LEVEL_1: 'administrative_area_level_1',
  COUNTRY: 'country',
  POSTAL_CODE: 'postal_code',
};

export const FIELD_NAME_BY_TYPE = {
  [GOOGLE_MAPS_ADDRESS_TYPE.STREET_NUMBER]: 'streetNumber',
  [GOOGLE_MAPS_ADDRESS_TYPE.ROUTE]: 'street',
  [GOOGLE_MAPS_ADDRESS_TYPE.LOCALITY]: 'city',
  [GOOGLE_MAPS_ADDRESS_TYPE.ADMINISTRATIVE_AREA_LEVEL_1]: 'state',
  [GOOGLE_MAPS_ADDRESS_TYPE.COUNTRY]: 'country',
  [GOOGLE_MAPS_ADDRESS_TYPE.POSTAL_CODE]: 'zipCode',
};

export const GOOGLE_MAPS_GEOCODING_TYPE = {
  LAT_LNG: 'latlng',
  ADDRESS: 'address',
};
