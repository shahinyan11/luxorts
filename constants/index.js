import ROUTES from 'constants/routes';

export const API_HOST = process.env.NEXT_PUBLIC_API_HOST || 'http://localhost:3000';
export const API_VERSION = '/api/v1';
export const SESSION_REFRESH_ROUTE = `${API_HOST}${API_VERSION}/user_account/session`;

export const USER_ROLE = {
  USER: 'user',
  GUEST: 'guest',
};

export const USER_TYPE = {
  TRAVELLER: 'traveller_account',
  HOST: 'host_account',
};

export const USER_VERIFIED_STATUS = {
  UNVERIFIED: 'unverified',
  EMAIL_VERIFIED: 'email_verified',
  PHONE_VERIFIED: 'phone_verified',
};

export const USER_LOGIN_REDIRECT_BY_STATUS = {
  [USER_VERIFIED_STATUS.UNVERIFIED]: ROUTES.EMAIL_VERIFICATION.PATH,
  [USER_VERIFIED_STATUS.EMAIL_VERIFIED]: ROUTES.PHONE_VERIFICATION.PATH,
  [USER_VERIFIED_STATUS.PHONE_VERIFIED]: ROUTES.INDEX.PATH,
};

export const DATE_FORMAT = 'DD/MM/YYYY';

export const DATE_FORMAT_WITH_TIME = 'MM/DD/YYYY [at] HH:MM A';

export const EMAIL_PLACEHOLDER = 'jeremy.bogan@gmail.com';

export const DEFAULT_COUNTRY_CODE = 'US';

export const MAX_INPUT_LENGTH = 255;

export const MAX_MESSAGE_LENGTH = 1000;

export const USER_ACCOUNT_ENTITIES = ['hostAccount', 'userProfile', 'travellerAccount'];

export const SPECIAL_CHARACTERS = /[`!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?~]/;

export const DIRECTION = {
  DOWN: 'down',
};

export const ANTD_CLASSNAME = {
  INPUT_NUMBER: 'ant-input-number-input',
};

export const APPROVED = 'APPROVED';

export const NOT_APPROVED = 'NOT_APPROVED';

export const RADIO_GROUP_PAID_OPTIONS = [
  {
    value: false,
    label: { id: 'shared.free' },
    className: 'ant-radio-wrapper--positive',
  },
  {
    value: true,
    label: { id: 'shared.paid' },
    className: 'ant-radio-wrapper--negative',
  },
];

export const FORM_DATA_CONFIG = {
  headers: {
    'Content-Type': 'multipart/form-data',
  },
};

export const TIME_OPTIONS = [
  { value: '6:00 AM', label: { id: 'shared.timeAm', values: { time: 6 } } },
  { value: '7:00 AM', label: { id: 'shared.timeAm', values: { time: 7 } } },
  { value: '8:00 AM', label: { id: 'shared.timeAm', values: { time: 8 } } },
  { value: '9:00 AM', label: { id: 'shared.timeAm', values: { time: 9 } } },
  { value: '10:00 AM', label: { id: 'shared.timeAm', values: { time: 10 } } },
  { value: '11:00 AM', label: { id: 'shared.timeAm', values: { time: 11 } } },
  { value: '12:00 PM', label: { id: 'shared.timePm', values: { time: 12 } } },
  { value: '1:00 PM', label: { id: 'shared.timePm', values: { time: 1 } } },
  { value: '2:00 PM', label: { id: 'shared.timePm', values: { time: 2 } } },
  { value: '3:00 PM', label: { id: 'shared.timePm', values: { time: 3 } } },
  { value: '4:00 PM', label: { id: 'shared.timePm', values: { time: 4 } } },
  { value: '5:00 PM', label: { id: 'shared.timePm', values: { time: 5 } } },
  { value: '6:00 PM', label: { id: 'shared.timePm', values: { time: 6 } } },
  { value: '7:00 PM', label: { id: 'shared.timePm', values: { time: 7 } } },
  { value: '8:00 PM', label: { id: 'shared.timePm', values: { time: 8 } } },
  { value: '9:00 PM', label: { id: 'shared.timePm', values: { time: 9 } } },
  { value: '10:00 PM', label: { id: 'shared.timePm', values: { time: 10 } } },
  { value: '11:00 PM', label: { id: 'shared.timePm', values: { time: 11 } } },
  { value: '12:00 AM', label: { id: 'shared.timeAm', values: { time: 12 } } },
  { value: '1:00 AM', label: { id: 'shared.timeAm', values: { time: 1 } } },
  { value: '2:00 AM', label: { id: 'shared.timeAm', values: { time: 2 } } },
  { value: '3:00 AM', label: { id: 'shared.timeAm', values: { time: 3 } } },
  { value: '4:00 AM', label: { id: 'shared.timeAm', values: { time: 4 } } },
  { value: '5:00 AM', label: { id: 'shared.timeAm', values: { time: 5 } } },
];

export const DAYS_IN = {
  WEEK: 7,
  MONTH: 31,
};

export const RECOVER = 'recover';

export const PAGE = {
  number: 1,
  size: 20,
};

export const AVATAR_SIZES = ['xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl', 'xxxl'];

export const SORT_DIRECTIONS = {
  ASCEND: 'ascend',
  DESCEND: 'descend',
};

export const INVITATION_STATUSES = {
  PENDING: 'pending',
  ACCEPTED: 'accepted',
};

export const HOME_PAGINATION = {
  number: 1,
  size: 3,
};

export const TRENDING_LOCATION_IMAGES = [
  '/images/trending-location-01.jpg',
  '/images/trending-location-02.jpg',
  '/images/trending-location-03.jpg',
  '/images/trending-location-04.jpg',
  '/images/trending-location-05.jpg',
];

export const PROPERTY_TYPE_IMAGES = [
  '/images/property-01.jpg',
  '/images/property-02.jpg',
  '/images/property-03.jpg',
  '/images/property-04.jpg',
];

export const SUPPORT_EMAIL = 'support@luxorts.com';

export const LANDING_TYPES = {
  POLICY: 'privacy_policy',
  TERMS: 'terms_of_service',
};
