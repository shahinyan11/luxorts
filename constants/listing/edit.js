import { compose, omit, values } from 'ramda';

import ROUTES from 'constants/routes';

// eslint-disable-next-line import/prefer-default-export
export const EDIT_LISTING_TABS = compose(
  values,
  omit(['KEY', 'PATH', 'META']),
)(ROUTES.DASHBOARD.LISTINGS);
