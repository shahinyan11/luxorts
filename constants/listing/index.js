import { v4 as uuid } from 'uuid';

import ROUTES from 'constants/routes';

export const LISTING_STEP_STATUS = {
  INITIAL: 'initial',
  ACCOMODATION: 'accomodation',
  LOCATION: 'location',
  LOCATION_MAP: 'location_map',
  AMENITIES: 'amenities',
  CAPABILITIES: 'capabilities',
  SERVICES: 'services',
  PHOTOS: 'photos',
  DESCRIPTION: 'description',
  HOUSE_RULES: 'house_rules',
  PREFERENCES: 'preferences',
  AVAILABILITY: 'availability',
  PRICING: 'pricing',
  FINISHED: 'finished',
};

export const LISTING_PATH_BY_STATUS = {
  [LISTING_STEP_STATUS.INITIAL]: ROUTES.ADD_NEW_LISTING.INDEX.PATH,
  [LISTING_STEP_STATUS.ACCOMODATION]: ROUTES.ADD_NEW_LISTING.ACCOMMODATION.PATH,
  [LISTING_STEP_STATUS.LOCATION]: ROUTES.ADD_NEW_LISTING.LOCATION.PATH,
  [LISTING_STEP_STATUS.LOCATION_MAP]: ROUTES.ADD_NEW_LISTING.LOCATION_MAP.PATH,
  [LISTING_STEP_STATUS.AMENITIES]: ROUTES.ADD_NEW_LISTING.AMENITIES.PATH,
  [LISTING_STEP_STATUS.CAPABILITIES]: ROUTES.ADD_NEW_LISTING.CAPABILITIES.PATH,
  [LISTING_STEP_STATUS.SERVICES]: ROUTES.ADD_NEW_LISTING.SERVICES.PATH,
  [LISTING_STEP_STATUS.PHOTOS]: ROUTES.ADD_NEW_LISTING.PHOTOS.PATH,
  [LISTING_STEP_STATUS.DESCRIPTION]: ROUTES.ADD_NEW_LISTING.DESCRIPTION.PATH,
  [LISTING_STEP_STATUS.HOUSE_RULES]: ROUTES.ADD_NEW_LISTING.HOUSE_RULES.PATH,
  [LISTING_STEP_STATUS.PREFERENCES]: ROUTES.ADD_NEW_LISTING.RESERVATION_PREFERENCES.PATH,
  [LISTING_STEP_STATUS.AVAILABILITY]: ROUTES.ADD_NEW_LISTING.AVAILABILITY.PATH,
  [LISTING_STEP_STATUS.PRICING]: ROUTES.ADD_NEW_LISTING.PRICING_AND_DISCOUNTS.PATH,
  [LISTING_STEP_STATUS.FINISHED]: ROUTES.ADD_NEW_LISTING.PUBLISH.PATH,
};

export const LISTING_PROGRESS_BY_STATUS = {
  [LISTING_STEP_STATUS.INITIAL]: 14,
  [LISTING_STEP_STATUS.ACCOMODATION]: 20,
  [LISTING_STEP_STATUS.LOCATION]: 26,
  [LISTING_STEP_STATUS.LOCATION_MAP]: 32,
  [LISTING_STEP_STATUS.AMENITIES]: 38,
  [LISTING_STEP_STATUS.CAPABILITIES]: 44,
  [LISTING_STEP_STATUS.SERVICES]: 50,
  [LISTING_STEP_STATUS.PHOTOS]: 56,
  [LISTING_STEP_STATUS.DESCRIPTION]: 62,
  [LISTING_STEP_STATUS.HOUSE_RULES]: 68,
  [LISTING_STEP_STATUS.PREFERENCES]: 74,
  [LISTING_STEP_STATUS.AVAILABILITY]: 80,
  [LISTING_STEP_STATUS.PRICING]: 86,
  [LISTING_STEP_STATUS.FINISHED]: 100,
};

export const LISTING_PLACE_TYPE = {
  ENTIRE_ROOM: 'entire_room',
  PRIVATE_PLACE: 'private_place',
  SHARED_ROOM: 'shared_room',
};

export const LISTING_PLACE_OPTIONS = [
  { value: LISTING_PLACE_TYPE.ENTIRE_ROOM, label: { id: 'shared.entirePlace' } },
  { value: LISTING_PLACE_TYPE.PRIVATE_PLACE, label: { id: 'shared.privateRoom' } },
  { value: LISTING_PLACE_TYPE.SHARED_ROOM, label: { id: 'shared.sharedRoom' } },
];

export const ACCOMMODATION_VALIDATION = {
  BASE: {
    MIN: 1,
    MAX: 16,
  },
  ARRANGEMENT: {
    MIN: 0,
    MAX: 16,
  },
};

export const BED_TYPE = {
  QUEEN: 'queen',
  SINGLE: 'single',
  DOUBLE: 'double',
  SOFA_BED: 'sofa_bed',
};

export const BED_NAME_BY_TYPE = {
  [BED_TYPE.DOUBLE]: { id: 'shared.double' },
  [BED_TYPE.QUEEN]: { id: 'shared.queen' },
  [BED_TYPE.SINGLE]: { id: 'shared.single' },
  [BED_TYPE.SOFA_BED]: { id: 'shared.sofaBed' },
};

export const LISTING_DEFAULT_BEDS = [
  {
    uuid: uuid(),
    kind: BED_TYPE.DOUBLE,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BED_TYPE.QUEEN,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BED_TYPE.SINGLE,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BED_TYPE.SOFA_BED,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
];

export const DEFAULT_BEDROOM_CONFIG = {
  defaultBeds: LISTING_DEFAULT_BEDS,
  customBeds: [],
  bedsForDeleting: [],
};

export const INITIAL_BEDROOM = {
  uuid: uuid(),
  ...DEFAULT_BEDROOM_CONFIG,
};

export const BATH_TYPE = {
  BATHTUB: 'bathtub',
  SHOWER: 'shower',
  JETTED_TUB: 'jetted_tub',
  DOUBLE_VANITY: 'double_vanity',
};

export const BATH_NAME_BY_TYPE = {
  [BATH_TYPE.BATHTUB]: { id: 'shared.bathtub' },
  [BATH_TYPE.SHOWER]: { id: 'shared.shower' },
  [BATH_TYPE.JETTED_TUB]: { id: 'shared.jettedTub' },
  [BATH_TYPE.DOUBLE_VANITY]: { id: 'shared.doubleVanity' },
};

export const LISTING_DEFAULT_BATHS = [
  {
    uuid: uuid(),
    kind: BATH_TYPE.BATHTUB,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BATH_TYPE.SHOWER,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BATH_TYPE.JETTED_TUB,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
  {
    uuid: uuid(),
    kind: BATH_TYPE.DOUBLE_VANITY,
    amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN,
  },
];

export const DEFAULT_BATHROOM_CONFIG = {
  defaultBaths: LISTING_DEFAULT_BATHS,
  customBaths: [],
  bathsForDeleting: [],
};

export const INITIAL_BATHROOM = {
  uuid: uuid(),
  ...DEFAULT_BATHROOM_CONFIG,
};

export const LISTING_STATUS = {
  IN_DRAFT: 'in_draft',
  ON_MODERATION: 'on_moderation',
  UNPUBLISHED: 'unpublished',
  PUBLISHED: 'published',
  BLOCKED: 'blocked',
  DELETED: 'deleted',
};

export const TIME_PERIOD_KIND = {
  AVAILABLE: 'available',
  BLOCKED: 'blocked',
  BOOKING: 'booking',
  PREPARATION: 'preparation',
};

export const TIME_PERIOD_DESCRIPTION_BY_KIND = {
  [TIME_PERIOD_KIND.BLOCKED]: { id: 'shared.blocked' },
  [TIME_PERIOD_KIND.PREPARATION]: { id: 'shared.preparationTime' },
  [TIME_PERIOD_KIND.BOOKING]: { id: 'shared.booked' },
};

export const TIME_PERIOD_ICON_BY_KIND = {
  [TIME_PERIOD_KIND.BLOCKED]: 'clock',
  [TIME_PERIOD_KIND.PREPARATION]: 'clock',
  [TIME_PERIOD_KIND.BOOKING]: 'clock',
};

export const LISTING_IMAGE_PLACEHOLDER = '/images/housing-preview-01.png';
