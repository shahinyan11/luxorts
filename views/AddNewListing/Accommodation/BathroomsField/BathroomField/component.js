import { Button, Form } from 'antd';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { ACCOMMODATION_VALIDATION, BATH_NAME_BY_TYPE } from 'constants/listing';
import isPresent from 'utils/isPresent';

import InputNumberField from 'views/shared/form/InputNumberField';
import InputField from 'views/shared/form/InputField';
import GradientButton from 'views/shared/GradientButton';
import SvgIcon from 'views/shared/SvgIcon';

import useContainer from './hook';

const BathroomField = ({ index, handleAddCustomBath, handleRemoveCustomBathCreator }) => {
  const { field, isEdit, bathsList, onEditHandler } = useContainer({ index });
  const hasBaths = isPresent(bathsList);

  return (
    <div className="new-listing__add mb-24 pb-8">
      <div className="new-listing__add-header d-flex justify-content-space-between align-item-center mb-8">
        <h3 className="new-listing__add-title mb-0">
          <FormattedMessage id="shared.bathroomNumber" values={{ number: index + 1 }} />
        </h3>
        {!isEdit && (
          <Button onClick={onEditHandler} htmlType="button" className="main-link">
            {hasBaths && <FormattedMessage id="shared.editBaths" />}
            {!hasBaths && <FormattedMessage id="shared.addBaths" />}
          </Button>
        )}
      </div>
      <p
        className={classNames(
          'new-listing__add-info mb-16',
          hasBaths && 'new-listing__add-info--filled',
        )}
      >
        {hasBaths && bathsList.join(', ')}
        {!hasBaths && (
          <FormattedMessage
            id="shared.bathsAmount"
            values={{ amount: ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN }}
          />
        )}
      </p>
      {isEdit && (
        <>
          <div className="new-listing__add-options max-width-364 mb-8">
            {field.value.defaultBaths.map((bath, bathIdx) => (
              <InputNumberField
                key={bath.id || bath.uuid}
                name={`bathrooms[${index}].defaultBaths[${bathIdx}].amount`}
                formItemClassName="mb-8 ant-form-item--light"
                addonBefore={<FormattedMessage {...BATH_NAME_BY_TYPE[bath.kind]} />}
                min={ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN}
                max={ACCOMMODATION_VALIDATION.ARRANGEMENT.MAX}
              />
            ))}
          </div>
          {field.value.customBaths.map((bath, bathIdx) => (
            <div key={bath.id || bath.uuid} className="d-flex">
              <Form.Item className="mb-16 ant-form-item--edit-number width-364">
                <InputField
                  name={`bathrooms[${index}].customBaths[${bathIdx}].kind`}
                  placeholder={{ id: 'shared.enterBathTitle' }}
                  withFormItem={false}
                />
                <InputNumberField
                  name={`bathrooms[${index}].customBaths[${bathIdx}].amount`}
                  min={ACCOMMODATION_VALIDATION.ARRANGEMENT.MIN}
                  max={ACCOMMODATION_VALIDATION.ARRANGEMENT.MAX}
                  withFormItem={false}
                />
              </Form.Item>
              <GradientButton
                onClick={handleRemoveCustomBathCreator({
                  bathroomIdx: index,
                  index: bathIdx,
                  deletedItem: bath,
                })}
                className="main-btn--icon ml-8"
                variant="tertiary"
                addonAfter={<SvgIcon icon="cross" className="icon-right" />}
              />
            </div>
          ))}
          <Button onClick={handleAddCustomBath} htmlType="button" className="main-link mb-24 mt-8">
            <FormattedMessage id="shared.addAnotherBath" />
          </Button>
          <GradientButton
            onClick={onEditHandler}
            className="min-width-140 mb-16"
            variant="secondary"
            text={{ id: 'shared.done' }}
          />
        </>
      )}
    </div>
  );
};

BathroomField.propTypes = {
  index: PropTypes.number.isRequired,
  handleAddCustomBath: PropTypes.func.isRequired,
  handleRemoveCustomBathCreator: PropTypes.func.isRequired,
};

export default BathroomField;
