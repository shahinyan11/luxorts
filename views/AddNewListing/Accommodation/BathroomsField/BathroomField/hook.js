import { useCallback, useEffect, useMemo, useState } from 'react';
import { useField } from 'formik';
import { useIntl } from 'react-intl';

import { ANTD_CLASSNAME } from 'constants';
import { BATH_NAME_BY_TYPE } from 'constants/listing';

import isPresent from 'utils/isPresent';
import disableHtmlElementsByClassName from 'utils/disableHtmlElementsByClassName';

function useContainer({ index }) {
  const [field] = useField(`bathrooms[${index}]`);
  const [isEdit, setIsEdit] = useState(false);
  const intl = useIntl();

  /**
   * Baths list
   * @type {array}
   */
  const bathsList = useMemo(() => {
    const list = [];

    [...field.value.defaultBaths, ...field.value.customBaths].forEach(({ amount, kind }) => {
      if (amount > 0 && isPresent(kind)) {
        const message = BATH_NAME_BY_TYPE[kind];
        const prefix = isPresent(message) ? intl.formatMessage(message) : kind;

        list.push(`${amount} ${prefix.toLowerCase()}`);
      }
    });

    return list;
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [field.value.defaultBaths, field.value.customBaths]);

  /**
   * Handle edit
   */
  const onEditHandler = useCallback(() => {
    setIsEdit(!isEdit);
  }, [isEdit]);

  /**
   * Disable antd inputs
   */
  const disableInputs = () => {
    if (isEdit || isPresent(field.value.customBaths)) {
      disableHtmlElementsByClassName(ANTD_CLASSNAME.INPUT_NUMBER);
    }
  };

  /**
   * Lifecycle
   */
  useEffect(disableInputs, [isEdit, field.value.customBaths]);

  return {
    field,
    isEdit,
    bathsList,
    onEditHandler,
    disableInputs,
  };
}

export default useContainer;
