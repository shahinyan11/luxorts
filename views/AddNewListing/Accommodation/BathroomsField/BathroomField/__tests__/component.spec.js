import { shallow } from 'enzyme';

import { DEFAULT_BATHROOM_CONFIG } from 'constants/listing';

import useContainer from '../hook';
import BathroomField from '../component';

const mockedHookData = {
  field: {
    value: DEFAULT_BATHROOM_CONFIG,
  },
  isEdit: false,
  bathsList: [],
  onEditHandler: jest.fn(),
};
jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('BathroomField component', () => {
  const props = {
    index: 0,
    handleAddCustomBath: jest.fn(),
    handleRemoveCustomBathCreator: jest.fn(() => jest.fn()),
  };

  let component = shallow(<BathroomField {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isEdit equals true', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      isEdit: true,
    });

    component = shallow(<BathroomField {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isEdit equals true and customBaths are present', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      isEdit: true,
      field: {
        value: {
          ...DEFAULT_BATHROOM_CONFIG,
          customBaths: [{ id: 'bath-1' }],
        },
      },
    });

    component = shallow(<BathroomField {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when bathsList is present', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      bathsList: ['1 bath'],
    });

    component = shallow(<BathroomField {...props} />);

    expect(component).toMatchSnapshot();
  });
});
