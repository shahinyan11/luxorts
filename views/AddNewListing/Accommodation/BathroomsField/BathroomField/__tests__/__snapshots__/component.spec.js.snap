// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BathroomField component matches snapshot 1`] = `
<div
  className="new-listing__add mb-24 pb-8"
>
  <div
    className="new-listing__add-header d-flex justify-content-space-between align-item-center mb-8"
  >
    <h3
      className="new-listing__add-title mb-0"
    >
      <Memo(FormattedMessage)
        id="shared.bathroomNumber"
        values={
          Object {
            "number": 1,
          }
        }
      />
    </h3>
    <Button
      className="main-link"
      htmlType="button"
      onClick={[MockFunction]}
    >
      <Memo(FormattedMessage)
        id="shared.addBaths"
      />
    </Button>
  </div>
  <p
    className="new-listing__add-info mb-16"
  >
    <Memo(FormattedMessage)
      id="shared.bathsAmount"
      values={
        Object {
          "amount": 0,
        }
      }
    />
  </p>
</div>
`;

exports[`BathroomField component matches snapshot when bathsList is present 1`] = `
<div
  className="new-listing__add mb-24 pb-8"
>
  <div
    className="new-listing__add-header d-flex justify-content-space-between align-item-center mb-8"
  >
    <h3
      className="new-listing__add-title mb-0"
    >
      <Memo(FormattedMessage)
        id="shared.bathroomNumber"
        values={
          Object {
            "number": 1,
          }
        }
      />
    </h3>
    <Button
      className="main-link"
      htmlType="button"
      onClick={[MockFunction]}
    >
      <Memo(FormattedMessage)
        id="shared.editBaths"
      />
    </Button>
  </div>
  <p
    className="new-listing__add-info mb-16 new-listing__add-info--filled"
  >
    1 bath
  </p>
</div>
`;

exports[`BathroomField component matches snapshot when isEdit equals true 1`] = `
<div
  className="new-listing__add mb-24 pb-8"
>
  <div
    className="new-listing__add-header d-flex justify-content-space-between align-item-center mb-8"
  >
    <h3
      className="new-listing__add-title mb-0"
    >
      <Memo(FormattedMessage)
        id="shared.bathroomNumber"
        values={
          Object {
            "number": 1,
          }
        }
      />
    </h3>
  </div>
  <p
    className="new-listing__add-info mb-16"
  >
    <Memo(FormattedMessage)
      id="shared.bathsAmount"
      values={
        Object {
          "amount": 0,
        }
      }
    />
  </p>
  <div
    className="new-listing__add-options max-width-364 mb-8"
  >
    <InputNumberField
      addonBefore={
        <Memo(FormattedMessage)
          id="shared.bathtub"
        />
      }
      formItemClassName="mb-8 ant-form-item--light"
      key="uuid/v4"
      max={16}
      min={0}
      name="bathrooms[0].defaultBaths[0].amount"
      withFormItem={true}
    />
    <InputNumberField
      addonBefore={
        <Memo(FormattedMessage)
          id="shared.shower"
        />
      }
      formItemClassName="mb-8 ant-form-item--light"
      key="uuid/v4"
      max={16}
      min={0}
      name="bathrooms[0].defaultBaths[1].amount"
      withFormItem={true}
    />
    <InputNumberField
      addonBefore={
        <Memo(FormattedMessage)
          id="shared.jettedTub"
        />
      }
      formItemClassName="mb-8 ant-form-item--light"
      key="uuid/v4"
      max={16}
      min={0}
      name="bathrooms[0].defaultBaths[2].amount"
      withFormItem={true}
    />
    <InputNumberField
      addonBefore={
        <Memo(FormattedMessage)
          id="shared.doubleVanity"
        />
      }
      formItemClassName="mb-8 ant-form-item--light"
      key="uuid/v4"
      max={16}
      min={0}
      name="bathrooms[0].defaultBaths[3].amount"
      withFormItem={true}
    />
  </div>
  <Button
    className="main-link mb-24 mt-8"
    htmlType="button"
    onClick={[MockFunction]}
  >
    <Memo(FormattedMessage)
      id="shared.addAnotherBath"
    />
  </Button>
  <GradientButton
    addonAfter={null}
    addonBefore={null}
    className="min-width-140 mb-16"
    onClick={[MockFunction]}
    text={
      Object {
        "id": "shared.done",
      }
    }
    variant="secondary"
  />
</div>
`;

exports[`BathroomField component matches snapshot when isEdit equals true and customBaths are present 1`] = `
<div
  className="new-listing__add mb-24 pb-8"
>
  <div
    className="new-listing__add-header d-flex justify-content-space-between align-item-center mb-8"
  >
    <h3
      className="new-listing__add-title mb-0"
    >
      <Memo(FormattedMessage)
        id="shared.bathroomNumber"
        values={
          Object {
            "number": 1,
          }
        }
      />
    </h3>
  </div>
  <p
    className="new-listing__add-info mb-16"
  >
    <Memo(FormattedMessage)
      id="shared.bathsAmount"
      values={
        Object {
          "amount": 0,
        }
      }
    />
  </p>
  <div
    className="new-listing__add-options max-width-364 mb-8"
  >
    <InputNumberField
      addonBefore={
        <Memo(FormattedMessage)
          id="shared.bathtub"
        />
      }
      formItemClassName="mb-8 ant-form-item--light"
      key="uuid/v4"
      max={16}
      min={0}
      name="bathrooms[0].defaultBaths[0].amount"
      withFormItem={true}
    />
    <InputNumberField
      addonBefore={
        <Memo(FormattedMessage)
          id="shared.shower"
        />
      }
      formItemClassName="mb-8 ant-form-item--light"
      key="uuid/v4"
      max={16}
      min={0}
      name="bathrooms[0].defaultBaths[1].amount"
      withFormItem={true}
    />
    <InputNumberField
      addonBefore={
        <Memo(FormattedMessage)
          id="shared.jettedTub"
        />
      }
      formItemClassName="mb-8 ant-form-item--light"
      key="uuid/v4"
      max={16}
      min={0}
      name="bathrooms[0].defaultBaths[2].amount"
      withFormItem={true}
    />
    <InputNumberField
      addonBefore={
        <Memo(FormattedMessage)
          id="shared.doubleVanity"
        />
      }
      formItemClassName="mb-8 ant-form-item--light"
      key="uuid/v4"
      max={16}
      min={0}
      name="bathrooms[0].defaultBaths[3].amount"
      withFormItem={true}
    />
  </div>
  <div
    className="d-flex"
    key="bath-1"
  >
    <FormItem
      className="mb-16 ant-form-item--edit-number width-364"
    >
      <InputField
        asComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "Group": [Function],
            "Password": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "action": "click",
                "iconRender": [Function],
                "visibilityToggle": true,
              },
              "render": [Function],
            },
            "Search": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "TextArea": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "render": [Function],
          }
        }
        name="bathrooms[0].customBaths[0].kind"
        numbersOnly={false}
        placeholder={
          Object {
            "id": "shared.enterBathTitle",
          }
        }
        withFormItem={false}
      />
      <InputNumberField
        max={16}
        min={0}
        name="bathrooms[0].customBaths[0].amount"
        withFormItem={false}
      />
    </FormItem>
    <GradientButton
      addonAfter={
        <SvgIcon
          className="icon-right"
          icon="cross"
        />
      }
      addonBefore={null}
      className="main-btn--icon ml-8"
      onClick={[MockFunction]}
      variant="tertiary"
    />
  </div>
  <Button
    className="main-link mb-24 mt-8"
    htmlType="button"
    onClick={[MockFunction]}
  >
    <Memo(FormattedMessage)
      id="shared.addAnotherBath"
    />
  </Button>
  <GradientButton
    addonAfter={null}
    addonBefore={null}
    className="min-width-140 mb-16"
    onClick={[MockFunction]}
    text={
      Object {
        "id": "shared.done",
      }
    }
    variant="secondary"
  />
</div>
`;
